packages/blockchain-wallet-v4-frontend/src/plugin/internal/background.ts

```diff
@@ -32,107 +32,106 @@ chrome.runtime.onConnect.addListener(async (port: chrome.runtime.Port) => {
+                data: e.message,
+                type: ConnectionEvents.Error
+              })
             }
-          } catch (e) {
-            await port.postMessage({
-              data: e.message,
-              type: ConnectionEvents.Error
-            })
-          }
-          break
-        case SupportedRPCMethods.SignTransaction:
-          await chrome.runtime.onMessage.addListener(listener)
-          try {
-            const isConnected = await isDomainConnected(metadata.origin)
-            if (isSessionActive) {
+            break
+          case SupportedRPCMethods.SignTransaction:
+            await chrome.runtime.onMessage.addListener(listener)
+            try {
               if (isConnected) {
                 openPopup(
                   `/plugin/sign-transaction?domain=${metadata.origin}&favicon=${
                     metadata.favicon
                   }&${transactionRequestToQueryParameters(transactionParams)}`
                 )
+              } else {
+                await port.postMessage({
+                  data: {
+                    code: 4100,
+                    message: messages.errors.notAuthorized()
+                  },
+                  type: ConnectionEvents.Error
+                })
               }
-            } else {
-              await chrome.storage.session.clear()
-              // eslint-disable-next-line
-              await chrome.tabs.create({ url: chrome.runtime.getURL('index-tab.html') }).catch((err) => console.log(err))
-            }
-          } catch (e) {
-            await port.postMessage({
-              data: e.message,
-              type: ConnectionEvents.Error
-            })
-          }
-          break
-        case SupportedRPCMethods.SignMessage:
-          await chrome.runtime.onMessage.addListener(listener)
-          try {
-            const isConnected = await isDomainConnected(metadata.origin)
-            if (!msg.params || !msg.params[0]) {
-              port.postMessage({
-                data: 'Error: Not binary data: undefined',
+            } catch (e) {
+              await port.postMessage({
+                data: e.message,
                 type: ConnectionEvents.Error
               })
-              break
             }
-            if (isSessionActive) {
+            break
+          case SupportedRPCMethods.SignMessage:
+            await chrome.runtime.onMessage.addListener(listener)
+            try {
               if (isConnected) {
                 openPopup(
-                  `/plugin/signature-request?domain=${metadata.origin}&favicon=${metadata.favicon}&message=${msg.params[0]}`
+                  `/plugin/signature-request?domain=${metadata.origin}&favicon=${metadata.favicon}&message=${transactionParams}`
                 )
+              } else {
+                await port.postMessage({
+                  data: {
+                    code: 4100,
+                    message: messages.errors.notAuthorized()
+                  },
```

Comment:
if we send error data like this then we have to change error handler in provider on line 168. Because it expects a string